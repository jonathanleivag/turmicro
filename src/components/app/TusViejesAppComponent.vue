<script setup lang="ts">
import { defineProps, onMounted, ref } from 'vue'
import Web3 from 'web3/dist/web3.min.js'

import { _viajesResponse, Turmicro } from '../../interfaces/ITurmicro'
import CardUiComponent from '../ui/CardUiComponent.vue'

interface IProps {
  account: string
  deployContract: Turmicro
  web3: Web3
}

const props = defineProps<IProps>()

const viajes = ref<_viajesResponse[]>([])

onMounted(() => {
  getViajes()
})

const getViajes = async () => {
  const totalViajes = +(await props.deployContract._totalViajes(props.account))
  for (let i = 0; i < totalViajes; i++) {
    const viajeCliente = await props.deployContract._vieajesCliente(
      props.account,
      i
    )
    viajes.value.push({
      nombre: viajeCliente.nombre,
      valor: +viajeCliente.valor,
      puntos: +viajeCliente.puntos
    })
  }
}
</script>

<template>
  <CardUiComponent title="Tus viejes">
    <ul class="card__list card__list--cliente">
      <li v-for="(viaje, index) in viajes" :key="viaje.nombre">
        <span>{{ index + 1 }}) {{ viaje.nombre }}</span>
      </li>
    </ul>
  </CardUiComponent>
</template>
