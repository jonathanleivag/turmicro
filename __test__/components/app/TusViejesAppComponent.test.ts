// TusViejesAppComponent

import { mount } from '@vue/test-utils'

import TusViejesAppComponent from '../../../src/components/app/TusViejesAppComponent.vue'

describe('TusViejesAppComponent', () => {
  const wrapper = mount(TusViejesAppComponent)

  test('Snapshot de TusViejesAppComponent', () => {
    expect(TusViejesAppComponent).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('El titulo tiene que ser Tus viejes', () => {
    expect(wrapper.find('h3').text()).toBe('Tus viejes')
  })
})
