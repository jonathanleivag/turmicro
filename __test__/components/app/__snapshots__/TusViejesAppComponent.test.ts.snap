// Vitest Snapshot v1

exports[`BalanceAppComponent > Snapshot de TusViejesAppComponent 1`] = `
"<div class=\\"card\\">
  <div class=\\"card__header\\">
    <h3 class=\\"card__title\\">Tus viejes</h3>
  </div>
  <div class=\\"card__body\\"></div>
</div>"
`;

exports[`TusViejesAppComponent > Snapshot de TusViejesAppComponent 1`] = `
"<div class=\\"card\\">
  <div class=\\"card__header\\">
    <h3 class=\\"card__title\\">Tus viejes</h3>
  </div>
  <div class=\\"card__body\\"></div>
</div>"
`;
